storage/oasis/client.go

```diff
@@ -19,10 +20,28 @@ const (
 
 // NewConsensusClient creates a new ConsensusClient.
 func NewConsensusClient(ctx context.Context, sourceConfig *config.SourceConfig) (*ConsensusClient, error) {
+	// If we are using purely file-backed indexer, do not connect to the node.
+	if sourceConfig.Cache != nil && sourceConfig.Cache.Consensus != nil && !sourceConfig.Cache.Consensus.QueryOnCacheMiss {
+		nodeApi, err := file.NewFileConsensusApiLite(sourceConfig.Cache.Consensus.CacheDir, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error instantiating cache-based consensusApi: %w", err)
+		}
+		return &ConsensusClient{
+			nodeApi: nodeApi,
+			network: sourceConfig.SDKNetwork(),
+		}, nil
+	}
+	var nodeApi nodeapi.ConsensusApiLite
```

Comment:
```suggestion

	// Create an API that connects to the real node, then wrap it in a caching layer.
```

I think we don't need the `var`? But mostly I just wanted to visually separate the two `QueryOnCacheMiss` alternatives, with a newline in between and a comment for each.

Reply:
Hm so we need the `var` or else golang's assumed type for `nodeApi` is `HistoryConsensusApiLite` which is too strict if we wrap/reassign it as a `FileConsensusApiLite` right after. I added the newline and comment; thanks!

Reply:
this is because we didn't go for that type-erased-constructors change last sprint :smiling_face_with_tear: